expr/func_aliasbymetric_test.go

```diff
@@ -0,0 +1,170 @@
+package expr
+
+import (
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+func TestAliasByMetricNameWithoutPeriods(t *testing.T) {
+	// Metric base same as the metric name
+	veryShortMetric := "veryShort"
+	veryShortBase := veryShortMetric
+
+	testAliasByMetric(
+		[]models.Series{
+			{ // No Function wrapper
+				Interval:   10,
+				QueryPatt:  veryShortMetric,
+				Target:     veryShortMetric,
+				Datapoints: getCopy(a),
+			},
+			{ // Function wrapper - single
+				Interval:   10,
+				QueryPatt:  veryShortMetric,
+				Target:     "functionBlah(" + veryShortMetric + ", funcValue1, funcValue2)",
+				Datapoints: getCopy(a),
+			},
+			{ // Function wrapper - multiple
+				Interval:   10,
+				QueryPatt:  veryShortMetric,
+				Target:     "functionBlah(functionBlahBlah(" + veryShortMetric + "),funcValue1, funcValue2)",
+				Datapoints: getCopy(a),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  veryShortBase,
+				Target:     veryShortBase,
+				Datapoints: getCopy(a),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  veryShortBase,
+				Target:     veryShortBase,
+				Datapoints: getCopy(a),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  veryShortBase,
+				Target:     veryShortBase,
+				Datapoints: getCopy(a),
+			},
+		},
+		t,
+	)
+}
+
+func TestAliasByMetricWithoutTags(t *testing.T) {
+	// Metric base equals string after the last period
+	shortMetric := "my.test.metric.short"
+	shortBase := "short"
+
+	testAliasByMetric(
+		[]models.Series{
+			{ // No Function wrapper
+				Interval:   10,
+				QueryPatt:  shortMetric,
+				Target:     shortMetric,
+				Datapoints: getCopy(a),
+			},
+			{ // Function wrapper - single
+				Interval:   10,
+				QueryPatt:  shortMetric,
+				Target:     "functionBlah(" + shortMetric + ", funcValue1, funcValue2)",
+				Datapoints: getCopy(a),
+			},
+			{ // Function wrapper - multiple
+				Interval:   10,
+				QueryPatt:  "a",
```

Comment:
should this be "a"?

Reply:
it's harmless, but not consistent. I'll change that.